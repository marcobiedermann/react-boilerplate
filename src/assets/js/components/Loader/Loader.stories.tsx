import { ComponentMeta, ComponentStoryObj } from '@storybook/react';
import Loader from '.';

export default {
  component: Loader,
} as ComponentMeta<typeof Loader>;

export const Default: ComponentStoryObj<typeof Loader> = {};
