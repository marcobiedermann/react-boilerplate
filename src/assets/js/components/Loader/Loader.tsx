import React, { ReactNode } from 'react';

function Loader(): JSX.Element {
  return <div>Loading â€¦</div>;
}

export default Loader;
