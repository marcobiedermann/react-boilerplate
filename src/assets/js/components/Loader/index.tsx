import React from 'react';

const Loader: React.FC = (props) => {
  return <div {...props} />;
};

export default Loader;
