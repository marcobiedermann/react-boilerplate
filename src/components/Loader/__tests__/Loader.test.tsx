import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import * as stories from '../Loader.stories';

const { Default } = composeStories(stories);

describe('loader', () => {
  it('should render correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
