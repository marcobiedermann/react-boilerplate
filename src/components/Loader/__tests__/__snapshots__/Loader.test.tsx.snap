// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`loader > should render correctly 1`] = `
<div>
  Loading â€¦
</div>
`;
